shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-transaction-spring/src/main/java/org/apache/shardingsphere/spring/transaction/ShardingTransactionTypeInterceptor.java

```diff
@@ -37,8 +38,16 @@
     public Object invoke(final MethodInvocation methodInvocation) throws Throwable {
         ShardingTransactionType shardingTransactionType = getAnnotation(methodInvocation);
         Objects.requireNonNull(shardingTransactionType, "could not found sharding transaction type annotation");
+        TransactionType preTransactionType = TransactionTypeHolder.get();
```

Comment:
it's better to add a unit test to verify this feature. current implementation will not work in multiple @ShardingTransactionType ?

Reply:
@ShardingTransactionType does not support multiple connections, and the TransactionType will be removed from the thread when the connection is closed.
E.g:
@ShardingTransactionType(value=BASE)
public void test (){
conn1 execute;(here is BASE)
conn2(or conn1) execute; (The expectation here is BASE, but is LOCAl)
}

Reply:
Got it, @ShardingTransactionType was designed together with @transactional before.
this feature will provide a raw jdbc way to support multiple connection transaction in shardingpshere.

appreciate you add a unit test to verify above code will be work fine.